tensorflow_lite_support/ios/task/vision/sources/TFLImageClassifier.m

```diff
@@ -0,0 +1,160 @@
+#import "tensorflow_lite_support/ios/task/processor/sources/TFLClassificationOptions+Helpers.h"
+#import "tensorflow_lite_support/ios/task/processor/utils/sources/TFLClassificationUtils.h"
+#import "tensorflow_lite_support/ios/task/vision/utils/sources/GMLImageUtils.h"
+
+#include "tensorflow_lite_support/c/task/vision/image_classifier.h"
+
+@interface TFLImageClassifier ()
+/** ImageClassifier backed by C API */
+@property(nonatomic) TfLiteImageClassifier *imageClassifier;
+@end
+
+@implementation TFLImageClassifierOptions
+@synthesize baseOptions;
+@synthesize classificationOptions;
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.baseOptions = [[TFLBaseOptions alloc] init];
+    self.baseOptions.modelFile = [[TFLExternalFile alloc] init];
+
+    self.baseOptions.computeSettings = [[TFLComputeSettings alloc] init];
+    self.baseOptions.computeSettings.cpuSettings = [[TFLCpuSettings alloc] init];
+    self.baseOptions.computeSettings.cpuSettings.numThreads = -1;
+
+    self.classificationOptions = [[TFLClassificationOptions alloc] init];
+    self.classificationOptions.maxResults = 5;
+    self.classificationOptions.scoreThreshold = 0;
+  }
+  return self;
+}
+
+- (nullable instancetype)initWithModelPath:(nonnull NSString *)modelPath {
+  self = [self init];
+  if (self) {
+    self.baseOptions.modelFile.filePath = modelPath;
+  }
+  return self;
+}
+
+@end
+
+@implementation TFLImageClassifier
+- (void)dealloc {
+  TfLiteImageClassifierDelete(_imageClassifier);
+}
+
+- (instancetype)initWithImageClassifier:(TfLiteImageClassifier *)imageClassifier {
+  self = [super init];
+  if (self) {
+    _imageClassifier = imageClassifier;
+  }
+  return self;
+}
+
++ (nullable instancetype)imageClassifierWithOptions:(nonnull TFLImageClassifierOptions *)options
+                                              error:(NSError **)error {
+  TfLiteImageClassifierOptions cOptions = TfLiteImageClassifierOptionsCreate();
+  if (![options.classificationOptions
+          copyClassificationOptionsToCClassificationOptions:&(cOptions.classification_options)
+                                                      error:error])
+    return nil;
+
+  [options.baseOptions copyBaseOptionsToCBaseOptions:&(cOptions.base_options)];
+
+  TfLiteSupportError *createClassifierError = nil;
+  TfLiteImageClassifier *imageClassifier =
+      TfLiteImageClassifierFromOptions(&cOptions, &createClassifierError);
+
+  [options.classificationOptions
+      deleteCStringArraysOfClassificationOptions:&(cOptions.classification_options)];
+
+  if (!imageClassifier) {
+    [TFLCommonUtils errorFromTfLiteSupportError:createClassifierError error:error];
+    TfLiteSupportErrorDelete(createClassifierError);
+    return nil;
+  }
+
+  return [[TFLImageClassifier alloc] initWithImageClassifier:imageClassifier];
+}
+
++ (nullable instancetype)imageClassifierWithModelPath:(nonnull NSString *)modelPath
```

Comment:
`imageClassifierWithModelPath` can be removed. The reason we have `FooWithModelPath` for text tasks is a purely historical reason. `imageClassifierWithOptions` is good enough.

Reply:
Done